### PR TITLE
Updating GetCases, GetMilestones, GetPlans, GetTests, GetResultsForRun to support pagination

### PR DESCRIPTION
**Pull request by renzors**
- Extending NewClient function to use Beta TestTail API (Beta API is which includes pagination at the moment, optional parameter)
- Creating sendRequestBeta function that contains pagination logic, it should be used in the bulk methods.
- Updating bulk methods that are being used by testradamus, to use the new sendRequestBeta function.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -19,12 +19,13 @@ type Client struct {
...
 	password   string
 	httpClient *http.Client
+	useBetaApi bool
 }
 
 // NewClient returns a new client
 // with the given credential
 // for the given testrail domain
-func NewClient(url, username, password string) (c *Client) {
+func NewClient(url, username, password string, useBetaApi ...bool) (c *Client) {
```
Comment (ohenak) [inline client.go:28]: Why using the `...` here?

File: client.go [hunk 2]
```diff
@@ -37,6 +38,10 @@ func NewClient(url, username, password string) (c *Client) {
 
 	c.httpClient = &http.Client{}
 
+	if len(useBetaApi) > 0 {
```
Comment (ohenak) [inline client.go:41]: Do you have a test case as an example on how to use this?

File: client.go (hunk 1 above)
Comment (renzors) [reply client.go:28]: That's the way I found to make useBetaApi parameter as optional, so people consuming this function won't be affected with the change. I saw the same approach in other functions that have optional parameters. e.g. https://github.com/renzors/testrail/blob/3ab3c62b1fdc7aa566e568972586b6cbb7e89569/milestone.go#L34

File: client.go (hunk 2 above)
Comment (renzors) [reply client.go:41]: Ah yes, it is not a test-case actually, I was testing in the main but I didn't pushed it. I will push a test-case soon.

Comment (renzors) [reply client.go:41]: I updated this test case to validate the default value (false) and the new value with parameter (true) https://github.com/renzors/testrail/blob/8ba6c275dd2fd0e8bddd7e15f776ba6176e22c45/client_test.go#L48

**Pull request merged**